tests/test_s3_to_json_s3.py

```diff
@@ -0,0 +1,477 @@
+                  "jsonSchema" : "http://example.org/schemas/v1/Foo.json",
+                  "metadata" : {
+                    "value" : 1
+                  }
+                }
+              ]
+            }
+          },
+          "properties" : {
+            "appName" : {
+              "type" : "string",
+              "description" : "Name of the app that built the archive."
+            },
+            "appVersion" : {
+              "type" : "string",
+              "description" : "Version of the app that built the archive."
+            },
+            "deviceInfo" : {
+              "type" : "string",
+              "description" : "Information about the specific device."
+            },
+            "deviceTypeIdentifier" : {
+              "type" : "string",
+              "description" : "Specific model identifier of the device."
+            },
+            "files" : {
+              "type" : "array",
+              "description" : "A list of the files included in this archive.",
+              "items" : {
+                "$ref" : "#/definitions/FileInfo"
+              }
+            }
+          },
+          "required" : [
+            "appName",
+            "appVersion",
+            "deviceInfo",
+            "deviceTypeIdentifier",
+            "files"
+          ]
+        }
+        return metadata_json_schema
+
+    def test_get_dataset_mapping(self, dataset_mapping_uri):
+        data_mapping = get_data_mapping(data_mapping_uri=dataset_mapping_uri)
+        assert isinstance(data_mapping, dict)
+
+    def test_get_schema_mapping(self, schema_mapping_uri):
+        schema_mapping = get_data_mapping(data_mapping_uri=schema_mapping_uri)
+        assert isinstance(schema_mapping, dict)
+
+    def test_get_archive(self, artifact_bucket, namespace):
+        archive_map = get_archive_map(archive_map_version="v4.4.1")
+        assert isinstance(archive_map, dict)
+
+    def test_update_sts_tokens(self):
+        synapse_data_folder_1 = "syn11111111"
+        sts_tokens = update_sts_tokens(
+                syn=MockSynapse(),
+                synapse_data_folder=synapse_data_folder_1,
+                sts_tokens={})
+        assert synapse_data_folder_1 in sts_tokens
+
+    def test_get_cached_json_schema(self):
+        json_schema = {"url": "url", "schema": "schema", "app_id": None, "assessment_id": None,
+                "assessment_revision": None, "file_name": None, "archive_map_version": None}
+        cached_json_schema = _get_cached_json_schema(
+            url="url",
+            json_schemas=[json_schema]
+        )
+        assert cached_json_schema == "schema"
+
+    def test_get_json_schema(self, archive_map):
+        file_metadata_no_schema = {
+                "assessment_id": "spelling",
+                "assessment_revision": 5,
+                "file_name": "jellybeanz.json",
+                "app_id": "mobile-toolbox"
+        }
+        json_schema_no_schema = get_json_schema(
```

Comment:
One thing to be more explicit is to do:

```
s3_to_json_s3.get_json_schema
```
to distinguish between the functions/fixtures you have within your test.

Reply:
Fixed.